src/Service/Startup.cs

```diff
@@ -153,8 +160,11 @@ public void ConfigureServices(IServiceCollection services)
 
             //Enable accessing HttpContext in RestService to get ClaimsPrincipal.
             services.AddHttpContextAccessor();
-
-            ConfigureAuthentication(services, configProvider);
+            services.AddTransient(implementationFactory: (serviceProvider) =>
+            {
+                ConfigureAuthentication(services, serviceProvider.GetRequiredService<RuntimeConfigProvider>());
```

Comment:
I don't understand why this is wrapped in a `services.Add..()` block

Reply:
this is old code, a new approach was taken, but to answer your question, it was to get access to the service provider.